### ISSUE TITLE
Feature request: publish a theme

### ISSUE THREAD

**Issue by andreacab** [Shopify/themekit#583]
I think it would be nice to be able to publish a theme from themekit. It seems the only two ways are:

1. Shopify REST Admin Api -> online store -> theme -> PUT /admin/themes/#{theme_id}.json
2. Shopify Admin UI, on the version desired, click "publish".

My bet is that it was probably thought of but I don't know why it wouldn't be implemented.

**Comment by tanema** [comment]
I am sure it has been thought of before, but it is nice to know someone is asking for it. I was already considering putting it into v1.0.0 and it's good to know some one else was thinking the same. v1.0.0 is in development but will not be release for a little while. I will update this issue on the progress.

**Comment by andreacab** [comment]
Cool thanks Tim.

**Comment by cameroncowden** [comment]
Would this require separate API permissions than Theme Files read/write?

**Comment by tanema** [comment]
No, all that is required is that single permission. Publish would just involve updating the theme's role.

**Comment by tanema** [comment]
It will not be for at least another month. This is not only for development but for people to appropriately adjust to no using the deprecated action in themekit because they will be removed in v1

**Comment by andreacab** [comment]
@tanema weekly but if we had more QA resources we would do it every time we have a change with the ideal of it being daily.

**Comment by andreacab** [comment]
I might be using themekit wrong, but it's hard to have the theme update get integrate well with our devops flow...

**Comment by tanema** [comment]
Okay @andreacab you can try out the new publish command by updating to the prerelease by running `theme update --version=v1.0.0-pre` Please let me know how it goes, and if you are happy with it I can close out this issue.

**Comment by andreacab** [comment]
@tanema the publish works great so far! thanks for the help!
